projects/telemetry/src/nmea.c

```diff
@@ -0,0 +1,33 @@
+#include "nmea.h"
+#include <stdbool.h>
+#include <string.h>
+
+static char s_hex[] = "0123456789ABCDEF";
+
+// out must be a pointer to at least 3 chars
+void prv_int_to_hex(uint8_t checksum, char *out) {
+  // we know that it'll be a 2 digit hex number since the checksum is 8-bit
+  out[0] = s_hex[checksum >> 4];
+  out[1] = s_hex[checksum & 15];
+  out[2] = '\0';
+}
+
+void nmea_compute_checksum(char *message, size_t message_len, char *out) {
+  uint8_t sum = 0;
+  for (uint8_t i = 1; message[i] != '*' && i < message_len; i++) {
+    sum ^= message[i];
+  }
+  prv_int_to_hex(sum, out);
```

Comment:
I know you mentioned that the reason the checksum function returns a string is for when you're building messages to send to the GPS, but is there a reason you don't just use `snprintf`? I assume you need to use something like it for building the rest of your command anyways.

Reply:
I meant to use `snprintf` in your command building and just use integers here.

Reply:
I could return an int here, but I would need to convert the int to hex when I'm building the command anyway, and in `compare_checksum`, I would need to either convert the int to hex, or convert the hex to int, so this seems like the best approach.

What I could do is return an int here, and then call `prv_int_to_hex` in `compare_checksum` instead, if that works.